builders/networkBuilder.go

```diff
@@ -0,0 +1,824 @@
+package builders
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"io"
+	"math/rand"
+	"net"
+	"os"
+	"strings"
+
+	"github.com/emicklei/dot"
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/tfexplorer"
+	"github.com/threefoldtech/tfexplorer/client"
+	"github.com/threefoldtech/tfexplorer/models/generated/workloads"
+	"github.com/threefoldtech/tfexplorer/schema"
+	"github.com/threefoldtech/zos/pkg"
+	"github.com/threefoldtech/zos/pkg/crypto"
+	"github.com/threefoldtech/zos/pkg/network/types"
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+)
+
+// NetworkBuilder is a struct that can build networks
+type NetworkBuilder struct {
+	workloads.Network
+
+	NodeID string
+	Bcdb   *client.Client
+	Mainui *tfexplorer.UserIdentity
+
+	AccessPoints []AccessPoint `json:"access_points,omitempty"`
+
+	// NetResources field override
+	NetResources []NetResource `json:"net_resources"`
+}
+
+// NetResource is the description of a part of a network local to a specific node
+type NetResource struct {
+	workloads.NetworkNetResource
+
+	// Public endpoints
+	PubEndpoints []net.IP `json:"pub_endpoints"`
+}
+
+// AccessPoint info for a network, defining a node which will act as the AP, and
+// the subnet which will be routed through it
+type AccessPoint struct {
+	// NodeID of the access point in the network
+	NodeID string `json:"node_id"`
+	// Subnet to be routed through this access point
+	Subnet      schema.IPRange `json:"subnet"`
+	WGPublicKey string         `json:"wg_public_key"`
+	IP4         bool           `json:"ip4"`
+}
+
+// NewNetworkBuilder creates a new network builder
+func NewNetworkBuilder(name string, iprange schema.IPRange) *NetworkBuilder {
```

Comment:
I don't see any argument to pass the explorer client to the builder. While it is required for the builder to work.
So I guess most of the method will just panic